examples/lines_bars_and_markers/linestyles.py

```diff
@@ -3,50 +3,65 @@
-     ('densely dotted',      (0, (1, 1))),
+linestyle_str = [
+     ('solid', 'solid'),       # Same as (0, ()) or '-'
+     ('dotted', 'dotted'),    # Same as (0, (1, 1)) or '.'
+     ('dashed', 'dashed'),    # Same as '--'
+     ('dashdot', 'dashdot')]  # Same as '-.'
 
-     ('loosely dashed',      (0, (5, 10))),
-     ('dashed',              (0, (5, 5))),
-     ('densely dashed',      (0, (5, 1))),
+linestyle_tuple = [
+     ('loosely dotted',        (0, (1, 10))),
+     ('densely dotted',        (0, (1, 1))),
 
-     ('loosely dashdotted',  (0, (3, 10, 1, 10))),
-     ('dashdotted',          (0, (3, 5, 1, 5))),
-     ('densely dashdotted',  (0, (3, 1, 1, 1))),
+     ('loosely dashed',        (0, (5, 10))),
+     ('densely dashed',        (0, (5, 1))),
+
+     ('loosely dashdotted',    (0, (3, 10, 1, 10))),
+     ('densely dashdotted',    (0, (3, 1, 1, 1))),
 
-     ('loosely dashdotdotted', (0, (3, 10, 1, 10, 1, 10))),
      ('dashdotdotted',         (0, (3, 5, 1, 5, 1, 5))),
-     ('densely dashdotdotted', (0, (3, 1, 1, 1, 1, 1)))])
+     ('loosely dashdotdotted', (0, (3, 10, 1, 10, 1, 10))),
+     ('densely dashdotdotted', (0, (3, 1, 1, 1, 1, 1)))]
 
 
-plt.figure(figsize=(10, 6))
-ax = plt.subplot(1, 1, 1)
+def simple_plot(ax, linestyles):
+    yticklabels = []
+    for i, (name, linestyle) in enumerate(linestyles):
+        ax.plot(X, Y+i, linestyle=linestyle, linewidth=1.5, color='black')
+        yticklabels.append(name)
+
+    ax.set(xticks=[], ylim=(-0.5, len(linestyles)-0.5),
+           yticks=np.arange(len(linestyles)), yticklabels=yticklabels)
+
+    # For each line style, add a text annotation with a small offset from
+    # the reference point (0 in Axes coords, y tick value in Data coords).
+    reference_transform = blended_transform_factory(ax.transAxes, ax.transData)
+    for i, (name, linestyle) in enumerate(linestyles):
+        ax.annotate(repr(linestyle), xy=(0.0, i), xycoords=reference_transform,
+                    xytext=(-6, -12), textcoords='offset points', color="blue",
+                    fontsize=8, ha="right", family="monospace")
 
 X, Y = np.linspace(0, 100, 10), np.zeros(10)
-for i, (name, linestyle) in enumerate(linestyles.items()):
-    ax.plot(X, Y+i, linestyle=linestyle, linewidth=1.5, color='black')
-
-ax.set_ylim(-0.5, len(linestyles)-0.5)
-plt.yticks(np.arange(len(linestyles)), linestyles.keys())
-plt.xticks([])
-
-# For each line style, add a text annotation with a small offset from
-# the reference point (0 in Axes coords, y tick value in Data coords).
-reference_transform = blended_transform_factory(ax.transAxes, ax.transData)
-for i, (name, linestyle) in enumerate(linestyles.items()):
-    ax.annotate(str(linestyle), xy=(0.0, i), xycoords=reference_transform,
-                xytext=(-6, -12), textcoords='offset points', color="blue",
-                fontsize=8, ha="right", family="monospace")
+fig, ax = plt.subplots(2, 1, gridspec_kw={'height_ratios': [1, 2]},
```

Comment:
Either use `fig, axs = ` or `fig, (ax1, ax2) = `. `ax` is canonically used for a single Axes only.